### PR TITLE
[llvm] Support real function which has scalar arguments

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #602

New TODOs:
1. Determine which passes real functions need.
2. (Maybe) let the function body be a serial offloaded statement to enable optimization passes
3. CFG dies when there are recursions. (Recursion works when we turn off CFG)

### CODE REVIEW

File: python/taichi/lang/impl.py [hunk 1]
```diff
@@ -226,34 +226,6 @@ def make_tensor_element_expr(_var, _indices, shape, stride):
                                           shape, stride))
 
 
-@taichi_scope
-def insert_expr_stmt_if_ti_func(ast_builder, func, *args, **kwargs):
```
Comment (k-ye) [inline python/taichi/lang/impl.py:230]: So, are we completely removing the experimental real function in this PR? (If so, I think the tests can be removed as well)

**Review by k-ye — APPROVED** [review]
LGTM!

File: python/taichi/lang/impl.py (hunk 1 above)
Comment (lin-hitonami) [reply python/taichi/lang/impl.py:230]: No, I will remove it in another PR, this is only a buggy function that prevents using `print` in the real functions. I merged this function with ASTTransformer.build_Expr.

**Comment by bobcao3** [comment]
"Let the function body be a serial offloaded statement to enable optimization passes" iirc in the analysis pass serial offloads will get all atomics demoted, that wouldn't be ideal...?

**Comment by lin-hitonami** [comment]
> "Let the function body be a serial offloaded statement to enable optimization passes" iirc in the analysis pass serial offloads will get all atomics demoted, that wouldn't be ideal...?

So maybe we shouldn't run demote_atomics pass on real functions. Let me check if there are other passes that should run on real functions that require offloaded statements.

**Review by ailzhang — APPROVED** [review]
LGTM!!

**Pull request merged**